internal/webservice/helpers.go

```diff
@@ -11,16 +11,67 @@ import (
 )
 
 const (
-	queryParamSubmissionStartTime = "submissionStartTime"
-	queryParamSubmissionEndTime   = "submissionEndTime"
-	queryParamGroups              = "groups"
-	queryParamLimit               = "limit"
-	queryParamOffset              = "offset"
-	queryParamUser                = "user"
-	queryParamTimestampStart      = "timestampStart"
-	queryParamTimestampEnd        = "timestampEnd"
+	queryParamSubmissionStartTime          = "submissionStartTime"
+	queryParamSubmissionEndTime            = "submissionEndTime"
+	queryParamGroups                       = "groups"
+	queryParamLimit                        = "limit"
+	queryParamOffset                       = "offset"
+	queryParamUser                         = "user"
+	queryParamTimestampStart               = "timestampStart"
+	queryParamTimestampEnd                 = "timestampEnd"
+	queryParamClusterID                    = "clusterId"
+	queryParamState                        = "state"
+	queryParamName                         = "name"
+	queryParamLastStateTransitionTimeStart = "lastStateTransitionTimeStart"
+	queryParamLastStateTransitionTimeEnd   = "lastStateTransitionTimeEnd"
 )
 
+func parsePartitionFilters(r *http.Request) (*repository.PartitionFilters, error) {
+	filters := repository.PartitionFilters{}
+	lastStateTransitionTimeStart, err := getLastStateTransitionTimeStartQueryParam(r)
+	if err != nil {
+		return nil, err
+	}
+	if lastStateTransitionTimeStart != nil {
+		filters.LastStateTransitionTimeStart = lastStateTransitionTimeStart
+	}
+	lastStateTransitionTimeEnd, err := getLastStateTransitionTimeEndQueryParam(r)
+	if err != nil {
+		return nil, err
+	}
+	if lastStateTransitionTimeEnd != nil {
+		filters.LastStateTransitionTimeEnd = lastStateTransitionTimeEnd
+	}
+	name := getNameQueryParam(r)
+	if name != "" {
```

Comment:
Why don't we just instantiate the filter with fields like the following:
```go
filters.Name = r.URL.Query().Get("name")
```
Since we are not re-using constants, I don't think they add anything to us here.

Reply:
Actually, we need to set `nil` in case of empty string `""`. Instead, if we set `filters.Name` with an empty string (when no query param is passed) that will cause querybuilder to give incorrect result.